src/tools/Reporting/Reporting/Reporter.cs

```diff
@@ -108,7 +110,8 @@ public string GetJson()
                 build,
                 os,
                 run,
-                tests
+                tests,
+                criticalErrors = HasCriticalValidationErrors
```

Comment:
I am not sure that this will work like we want. If I understand correctly, if any of the individual benchmarks have a failure, than all of the result json files will have this critical errors field marked true. However, parts of the run that occurred on another machine will have this value set to false, and then depending on which result json we end up processing last will dictate whether we mark that run as having failures or not. 

To my thinking we could either fix this, by storing this information at the test level, or making a change in the uploader to respect the fact that the critical error value has been set true. I think I would prefer storing this at the test level, as tests executed on another machine that did not have errors should not really be thought of as suspect in any way.

Reply:
HasCriticalValidationErrors is a field that is tied to the Summary object, not individual reports. So from what I can tell there is no way to associate it to a single test run.

Reply:
Here is the source code for `HasCriticalValidationErrors`: https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Reports/Summary.cs#L74

I think it makes sense we can't associate it to a single test run, since some of the validators apply to multiple test runs (e.g. ones that check if there are too many benchmarks in one category)

Reply:
OK, that makes sense, but we still will need to apply the attribute at the test level when we are putting it into the json, or we will still have the same issue of either clobbering the correct value for the critical error attribute, or treating every test from a run as having critical errors when they only occurred in a small subset of the tests.

Could solve this by just iterating over all of the tests in the GetJson method and adding this failure to the additional data fields that we already carry with each test.